reference/regexp.go

```diff
@@ -62,6 +62,14 @@ var (
 		capture(nameComponentRegexp,
 			optional(repeated(literal(`/`), nameComponentRegexp))))
 
+	// anchoredDomainRegexp is used to validate a domain value of reference.
+	anchoredDomainRegexp = anchored(domainRegexp)
+
+	// anchoredPathRegexp is used to validate a path value of reference.
+	anchoredPathRegexp = anchored(
+		capture(nameComponentRegexp,
+			optional(repeated(literal(`/`), nameComponentRegexp))))
```

Comment:
Should we define a `pathRegexp` that's used by `anchornedPathRegexp`, `NameRegexp`, and `anchoredNameRegexp`?